test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs

```diff
@@ -30,24 +31,18 @@ protected ApplicationTestFixture(string applicationName)
 
         public ILogger Logger { get; private set; }
 
-        public DeploymentResult CreateDeployment()
-        {
-            lock (_deploymentLock)
-            {
-                if (_deployer != null)
-                {
-                    return _deploymentResult;
-                }
+        public ILoggerFactory LoggerFactory { get; private set; }
 
-                Logger = CreateLogger();
-                var deploymentParameters = GetDeploymentParameters();
-                var deployer = ApplicationDeployerFactory.Create(deploymentParameters, Logger);
-                _deploymentResult = deployer.Deploy();
+        public async Task<DeploymentResult> CreateDeploymentAsync()
```

Comment:
We might need the lock as yet. Do we run the tests in parallel or does the deployer manage concurrency?